python/tests/test_zarr.py

```diff
@@ -10,58 +10,129 @@
 zarr = pytest.importorskip("zarr")
```

Comment:
The top-level `cupy` importorskip will mean that if cupy is not available we will never run numpy-based tests. Is that what you want?